lib/elixir/lib/string.ex

```diff
@@ -709,6 +709,34 @@ defmodule String do
     {left, right}
   end
 
+  @doc ~S"""
+  Counts the number of occurrences of a pattern in a string.
+
+  ## Examples
+
+      iex> String.count("hello world", "o")
+      2
+
+      iex> String.count("hello world", "l")
+      3
+
+      iex> String.count("hello world", "x")
+      0
+
+      iex> String.count("hello world", ~r/o/)
+      2
+
+  """
+  @spec count(t, pattern | Regex.t()) :: non_neg_integer
```

Comment:
Since the type is `pattern`, could we perhaps include some doctest examples with a compiled pattern for instance?